runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1755,12 +1768,27 @@ TR::Node *TR_VectorAPIExpansion::fromBitsCoercedIntrinsicHandler(TR_VectorAPIExp
    {
    TR::Compilation *comp = opt->comp();
 
+   TR::Node *broadcastTypeNode = node->getChild(4);
+
+   if (!broadcastTypeNode->getOpCode().isLoadConst())
+      {
+      if (opt->_trace) traceMsg(comp, "Unknown broadcast type in node %p\n", node);
+      return NULL;
+      }
+
+   int32_t broadcastType = broadcastTypeNode->get32bitIntegralValue();
+
+   TR_ASSERT_FATAL(broadcastType == MODE_BROADCAST || broadcastType == MODE_BITS_COERCED_LONG_TO_MASK,
+                   "Unexpected broadcast type in node %p\n", node);
+
+   bool mask = (broadcastType == MODE_BITS_COERCED_LONG_TO_MASK);
+
    if (mode == checkScalarization)
-      return node;
+      return mask ? NULL : node;
 
    if (mode == checkVectorization)
       {
-      TR::ILOpCodes splatsOpCode = TR::ILOpCode::createVectorOpCode(TR::vsplats,
+      TR::ILOpCodes splatsOpCode = TR::ILOpCode::createVectorOpCode(mask ? TR::mLongBitsToMask: TR::vsplats,
```

Comment:
You need to do the same thing when the mode is `doVectorization`.

Reply:
Fixed. Thanks!